src/uu/date/src/date.rs

```diff
@@ -305,11 +325,26 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
                     format!("{date_part} 00:00 {offset}")
                 };
                 parse_date(composed)
-            } else if let Some(total_hours) = military_tz_with_offset {
+            } else if let Some((total_hours, day_delta)) = military_tz_with_offset {
                 // Military timezone with optional hour offset
                 // Convert to UTC time: midnight + military_tz_offset + additional_hours
-                let date_part =
-                    strtime::format("%F", &now).unwrap_or_else(|_| String::from("1970-01-01"));
+
+                // When calculating a military timezone with an optional hour offset, midnight may
+                // be crossed in either direction. `day_delta` indicates wether the date remains
+                // the same, moves to the previous day, or advances to the next day.
+                let date_part = match day_delta {
+                    DayDelta::Same => {
+                        strtime::format("%F", &now).unwrap_or_else(|_| String::from("1970-01-01"))
+                    }
+                    DayDelta::Next => now
+                        .tomorrow()
+                        .and_then(|d| strtime::format("%F", &d))
+                        .unwrap_or_else(|_| String::from("1970-01-01")),
+                    DayDelta::Previous => now
+                        .yesterday()
+                        .and_then(|d| strtime::format("%F", &d))
+                        .unwrap_or_else(|_| String::from("1970-01-01")),
+                };
```

Comment:
can probably simplified with something like:

```suggestion
                let format_date = |date: Option<Zoned>| -> String {
                    date.and_then(|d| strtime::format("%F", &d))
                        .unwrap_or_else(|_| String::from("1970-01-01"))
                };
                let date_part = match day_delta {
                    DayDelta::Same => format_date(Some(now)),
                    DayDelta::Next => format_date(now.tomorrow()),
                    DayDelta::Previous => format_date(now.yesterday()),
                };
```